### PR TITLE
feat: minimal plugin setup

### PR DESCRIPTION
**Pull request by martijnvdbrug**
Minimal setup for new plugins according to this discussion: https://github.com/vendure-ecommerce/vendure/discussions/2039

image

### CODE REVIEW

File: test/e2e.spec.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  SimpleGraphQLClient,
+  SqljsInitializer,
+  testConfig,
+} from '@vendure/testing';
+import { TestServer } from '@vendure/testing/lib/test-server';
+import { ExamplePlugin } from '../src/example.plugin';
+import { initialData } from './initial-data';
+import path from 'path';
+
+describe('Limit variants per order plugin', function () {
```
Comment (michaelbromley) [inline test/e2e.spec.ts:14]: Copy paste artifact? I guess we just need a generic label.

**Review by michaelbromley — COMMENTED** [review]
I feel like we should keep the `/api` dir and put the resolver and the schema there (in `api-extensions.ts`). It is quite usual for the api layer to expand to several resolvers - shop, admin, and often entity field resolvers.

In general the pattern I use for virtually all my plugins, and what I think we should standardize on is like this:

```
- api (resolvers, schema defs)
- config (strategies, custom field defs etc)
- entity (typeorm entities)
- service (service layer - main business logic)
```

These directories cover almost everything you need in most plugins, and if we have a de-facto standard it makes it very easy to work with Vendure projects in general.

**Comment by martijnvdbrug** [comment]
> I feel like we should keep the `/api` dir and put the resolver and the schema there (in `api-extensions.ts`). It is quite usual for the api layer to expand to several resolvers - shop, admin, and often entity field resolvers.
> 
> In general the pattern I use for virtually all my plugins, and what I think we should standardize on is like this:
> 
> ```
> - api (resolvers, schema defs)
> - config (strategies, custom field defs etc)
> - entity (typeorm entities)
> - service (service layer - main business logic)
> ```
> 
> These directories cover almost everything you need in most plugins, and if we have a de-facto standard it makes it very easy to work with Vendure projects in general.

Ahh ok, so you put the schema and resolvers in the same file? That's good bundling, I'm going to start using that as well :-)

**Comment by michaelbromley** [comment]
Thanks for your work on this!

**Pull request merged**